sql/src/main/java/io/crate/analyze/repositories/RepositorySettingsModule.java

```diff
@@ -138,4 +98,28 @@ protected void configure() {
         typeSettingsBinder.addBinding(HDFS).toInstance(HDFS_SETTINGS);
         typeSettingsBinder.addBinding(S3).toInstance(S3_SETTINGS);
     }
+
+    private static Map<String, Setting> convertSettingListToMap(List<Setting> settingList) {
+        return settingList.stream().collect(Collectors.toMap(Setting::getKey, Function.identity()));
+    }
+
+    /**
+     * Copy and rename {@link Setting}s.
+     * For each Setting, a new Setting is created
+     * with it's key suffix acting as the new key
+     *
+     * eg. From Setting: s3.client.default.endpoint
+     * a new Setting will be created with same characteristics
+     * and new key name: endpoint
+     */
+    private static List<Setting> renameSettingsUsingSuffixAsKey(List<Setting> settingList) {
+        return settingList
+            .stream()
+            .map(s -> s.copyAndRename(k -> getSuffixOrInput((String) k)))
```

Comment:
string cast should not be needed here

Reply:
this was a surprise for me too - I didn't expect that a cast is in need as the `apply` method is templated.  
However, it expects an `Object` for some reason. Without the cast I get an error: "incompatible types: java.lang.Object cannot be converted to java.lang.String"

What am I missing here?

Reply:
type seemed to be inferred from Setting<T> :/

Reply:
> type seemed to be inferred from Setting :/

yeah, I guess I could change all methods returning `List<Setting>` to return `List<Setting<T>>`
Not sure if it's worth it to be honest